### ISSUE TITLE
Sign In Apple Compatibility is not added in iOS build

### ISSUE THREAD

**Issue by gaurangbackrock** [lupidan/apple-signin-unity#213]
Unity 6000.0.17f1 version.

Error = Exception: Can't Add Sign In With Apple programmatically in this Unity version

When taking out the iOS build, Unity experiences the above error, and in Xcode, "Sign In Apple" Compatibility is not added and must be manually added.

Please can anyone help me?

**Comment by karl-** [comment]
It looks like the constructor signature was changed as of Unity 6000.x. Modifying `ProjectCapabilityManagerExtension.cs` got me running again, although I can't verify that it is actually working yet as I'm still fighting other fires on the upgrade process.

```diff
diff --git a/Packages/com.lupidan.apple-signin-unity-src/Editor/ProjectCapabilityManagerExtension.cs b/Packages/com.lupidan.apple-signin-unity-src/Editor/ProjectCapabilityManagerExtension.cs
index c1882a7f9d..e4f668a564 100644
--- a/Packages/com.lupidan.apple-signin-unity-src/Editor/ProjectCapabilityManagerExtension.cs
+++ b/Packages/com.lupidan.apple-signin-unity-src/Editor/ProjectCapabilityManagerExtension.cs
@@ -25,21 +25,29 @@ namespace AppleAuth.Editor
         {
             var managerType = typeof(ProjectCapabilityManager);
             var capabilityTypeType = typeof(PBXCapabilityType);
-            
+
             var projectField = managerType.GetField("project", NonPublicInstanceBinding);
             var targetGuidField = managerType.GetField("m_TargetGuid", NonPublicInstanceBinding);
             var entitlementFilePathField = managerType.GetField("m_EntitlementFilePath", NonPublicInstanceBinding);
             var getOrCreateEntitlementDocMethod = managerType.GetMethod("GetOrCreateEntitlementDoc", NonPublicInstanceBinding);
+
+#if UNITY_6000_0_OR_NEWER
             var constructorInfo = capabilityTypeType.GetConstructor(
                 PublicInstanceBinding,
                 null,
-                new[] {typeof(string), typeof(bool), typeof(string), typeof(bool)}, 
+                new[] { typeof(bool), typeof(string), typeof(bool) },
                 null);
-            
+#else
+            var constructorInfo = capabilityTypeType.GetConstructor(
+                PublicInstanceBinding,
+                null,
+                new[] {typeof(string), typeof(bool), typeof(string), typeof(bool)},
+                null);
+#endif
             if (projectField == null || targetGuidField == null  || entitlementFilePathField == null ||
                 getOrCreateEntitlementDocMethod == null || constructorInfo == null)
                 throw new Exception("Can't Add Sign In With Apple programatically in this Unity version");
-            
+
             var entitlementFilePath = entitlementFilePathField.GetValue(manager) as string;
             var entitlementDoc = getOrCreateEntitlementDocMethod.Invoke(manager, new object[] { }) as PlistDocument;
             if (entitlementDoc != null)
@@ -53,7 +61,11 @@ namespace AppleAuth.Editor
             if (project != null)
             {
                 var mainTargetGuid = targetGuidField.GetValue(manager) as string;
+#if UNITY_6000_0_OR_NEWER
+                var capabilityType = constructorInfo.Invoke(new object[] { true, string.Empty, true }) as PBXCapabilityType;
+#else
                 var capabilityType = constructorInfo.Invoke(new object[] { "com.apple.developer.applesignin.custom", true, string.Empty, true }) as PBXCapabilityType;
+#endif
 
                 var targetGuidToAddFramework = unityFrameworkTargetGuid;
                 if (targetGuidToAddFramework == null)

```

**Comment by gaurangbackrock** [comment]
Yes i already updated by the help of google & there is no error in iOS build, but still there is no apple compatibility added.

**Comment by GAGOSIAN84** [comment]
This happens to me.
Unity 2022.3.47f1
Sign in with Apple Unity Plugin v1.4.3
Xcode 16.0 (16A242d)

**Comment by nadir500** [comment]
This issue still persists on Unity 6000.0.23f1 which is the latest LTS by the date of this comment
any thoughts? @lupidan

**Comment by lupidan** [comment]
Hello @nadir500,

Yes, I have been aware for a few weeks about this issue in particular. But I have not been able to find time to sit down, test, and make a new version.

In the meantime, I think a possible solution is nicely provided by @karl- here:
https://github.com/lupidan/apple-signin-unity/pull/214

**Comment by lupidan** [comment]
Released v1.4.4 with Unity 6000 support

https://github.com/lupidan/apple-signin-unity/releases/tag/v1.4.4
